### PR TITLE
ET-4894 add prod sagemaker endpoint

### PR DESCRIPTION
**Pull request by Robert-Steiner**
## What 🕵️ 🔍

- adds sagemaker template
- adds sagemaker bench script
- adds sagemaker custom inference script
- adds just receipt to create a hugging face model data package with a custom inference script
- adds sagemaker pro endpoint for multilingual-e5-small
- the endpoint is not yet deployed in the vpc. i will do it in the tenant pr 
- moved normalization of embeddings into the python script 

## References 📝 🔗

- [Jira task](https://xainag.atlassian.net/browse/ET-4894)

## Reviewer checklist

*Reviewer agreement:*

* Reviewers shouldn't review a "Draft Pull Request", but rather wait for it to be marked as "Ready for Review".
* Reviewers do **not** commit or merge the pull request.
* If/when there are changes that need to be applied (applying Terraform or deploying Kubernetes manifests), the **assignee** is the one responsible for doing it.
* Use the labels `not-applied`, `partially-applied` and `fully-applied` so others know the current progress.
  * Remember to change labels accordingly once you start and finish to apply the changes.

### CODE REVIEW

File: b2b_personalization/infra/prod/eu-central-1/b2b/sagemaker_multilingual_e5_small/terragrunt.hcl [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    # {
+    #   initial_instance_count = 1
+    #   instance_type          = "ml.g4dn.xlarge"
+    #   variant_name           = "gpu"
+    # }
+  ]
+
+  # when using gpu kms needs to be turned off
+  # create_kms = false
+  # create_endpoint = false
```
Comment (joecks) [inline b2b_personalization/infra/prod/eu-central-1/b2b/sagemaker_multilingual_e5_small/terragrunt.hcl:76]: 🎨 I guess removing all the comments is fine, when we have the file in _env as reference

File: b2b_personalization/infra/prod/eu-central-1/b2b/sagemaker_multilingual_e5_small/terragrunt.hcl
Comment (joecks) [inline b2b_personalization/infra/prod/eu-central-1/b2b/sagemaker_multilingual_e5_small/terragrunt.hcl:1]: I understand that you wanna mark the model with the name of the dir, but wouldn't that be confusing when we always have to rename that directory?

Comment (Robert-Steiner) [reply b2b_personalization/infra/prod/eu-central-1/b2b/sagemaker_multilingual_e5_small/terragrunt.hcl:1]: the directory is actually a sagemaker endpoint, not just a model. added the name of the model in order to add other models later (if needed). if we want to change a model we can create a new endpoint and delete the old one

File: b2b_personalization/infra/prod/eu-central-1/b2b/sagemaker_multilingual_e5_small/terragrunt.hcl (hunk 1 above)
Comment (Robert-Steiner) [reply b2b_personalization/infra/prod/eu-central-1/b2b/sagemaker_multilingual_e5_small/terragrunt.hcl]: i refactored the code to reduce the code duplication

**Review by joecks — APPROVED** [review]

**Pull request merged**